File: inst/d3/scatterplot/scatterplot.js
// Setting the margin object
margin = {
  top: 40,
  left: 50,
  right: 30,
  bottom: 40,
};

// Setting the min and max of the x variable
let xmin = d3.min(data, (d) => d[options.x]);
let xmax = d3.max(data, (d) => d[options.x]);

// Setting the min and max of the y variable
let ymin = d3.min(data, (d) => d[options.y]);
let ymax = d3.max(data, (d) => d[options.y]);

// Setting the scale of the x variable
let x = d3
  .scaleLinear()
  .domain(d3.extent(data, d => d[options.x]))
  .range([margin.left, width - margin.right])
  .nice();

// Setting the scale of the y variable
let y = d3
  .scaleLinear()
  .domain(d3.extent(data, d => d[options.y]))
  .range([height - margin.top, margin.bottom])
  .nice();

// Setting the x-axis
xS = d3.axisBottom(x).ticks(options.xticks);

// Setting the y-axis
yS = d3.axisLeft(y).ticks(options.yticks);



// Rendering the x-axis
svg
  .append("g")
  .attr("transform", "translate(0," + (height - margin.bottom) + ")")
  .call(xS)

// Rendering the y-axis
svg
  .append("g")
  .attr("transform", "translate(" + margin.left + ", 0)")
  .call(yS)


// Rendering the x-axis title
svg
  .append("text")
  .attr("transform", "translate(" + width / 2 + " ," + (height - 5) + ")")
  .attr("dx", "0em")
  .style("text-anchor", "middle")
  .style("font-family", options.font)
  .style("font-size", options.xtitleFontSize)
  .text(options.xtitle)
  .style('fill', options.axisCol)

// Rendering the y-axis title
svg
  .append("text")
  .attr("transform", "translate(" + 0 + " ," + height / 2 + ") rotate(-90)")
  .attr("dy", "0.8em")
  .style("text-anchor", "middle")
  .style("font-size", options.ytitleFontSize)
  .style("font-family", options.font)
  .text(options.ytitle)
  .style('fill', options.axisCol)

// Rendering the chart title
svg
  .append("text")
  .attr("x", width / 2)
  .attr("y", margin.top / 2)
  .attr("text-anchor", "middle")
  .attr("dx", "0em")
  .style("font-size", options.titleFontSize)
  .style("font-family", options.font)
  .text(options.title)
  .style('fill', options.axisCol)


svg.selectAll(".tick line").attr("stroke", options.axisCol)
svg.selectAll(".tick text").style("fill", options.axisCol)
svg.selectAll("path.domain").attr("stroke", options.axisCol)


// Rendering the scatter plot
svg
  .style("background-color", options.bgcol)
  .selectAll("circle")
  .data(data)
  .enter()
  .append("circle")
  .attr("cx", (d) => x(d[options.x]))
  .attr("cy", (d) => y(d[options.y]))
  .attr("r", options.size)
  .attr("fill", options.col)
  .attr("opacity", options.opacity)
  .attr("stroke", options.stroke)
  .attr("stroke-width", options.strokeWidth);

File: R/scatterPlot.R

#' Create a scatter plot.
#'
#' @param data The data frame containing the quantitative variables.
#' @param x The x-variable to consider.
#' @param y The y-variable to consider.
#' @param col The color of the dots. Defaults to 'crimson'.
#' @param size The size of the dots. Defaults to 2.
#' @param xticks Optional. The number of x-axis ticks to consider.
#' @param yticks Optional. The number of y-axis ticks to consider.
#' @param xtitle Optional. the title of the x-axis.
#' @param xtitleFontSize The font size of the x-axis title. Defaults to 16.
#' @param ytitle Optional. The title of the y-axis.
#' @param ytitleFontSize The font size of the y-axis title. Defaults to 16.
#' @param title Optional. the title of the plot.
#' @param titleFontSize The font size of the plot title. Defaults to 22.
#' @param stroke Optional. the stroke color of the dots.
#' @param strokeWidth Optional. the stroke width of the dots.
#' @param font The font family to consider for the titles. Defaults to
#' "Verdana, Geneva, Tahoma, sans-serif".
#' @param bgcol The background color of the SVG. Defaults to "#CAD0D3" HEX color.
#' @param opacity The color opacity of the dots (from 0 to 1). Defaults to 1.
#' @param axisCol the color of the x and y axis. It includes the ticks, the labels and titles.
#' Defaults to 'black'.
#' @param width Optional. the width of the SVG output.
#' @param height Optional. the height of the SVG output.
#'
#' @return A SVG scatter plot.
#' @export
#'
#' @examples
#'
#'scatterPlot(
#'   data = mtcars,
#'   x = "mpg",
#'   y = "wt"
#'   )
#'

scatterPlot <- function(
                        data,
                        x,
                        y,
                        col = "crimson",
                        size = 2,
                        xticks = NULL,
                        yticks = NULL,
                        xtitle = NULL,
                        xtitleFontSize = 16,
                        ytitle = NULL,
                        ytitleFontSize = 16,
                        title = NULL,
                        titleFontSize = 22,
                        stroke = NULL,
                        strokeWidth = NULL,
                        font = "Verdana, Geneva, Tahoma, sans-serif",
                        bgcol = "#CAD0D3",
                        opacity = 1,
                        axisCol = "black",
                        width = NULL,
                        height = NULL
                        ) {

  if (is.null(data[[x]]) || is.null(data[[y]])) {
    stop("Please check that x and y belong to the specified data frame")
  }

  if (grepl(";", font)) {
    stop("please remove the ';' character from your font argument")
  }

  r2d3::r2d3(
    data = data,
    script = system.file(
      "d3/scatterplot/scatterplot.js",
       package = "ddplot"
    ),
    options = list(
      x = x,
      y = y,
      col = col,
      size = size,
      xticks = xticks,
      yticks = yticks,
      xtitle = xtitle,
      ytitle = ytitle,
      title = title,
      font = font,
      stroke = stroke,
      strokeWidth = strokeWidth,
      width = width,
      height = height,
      xtitleFontSize = xtitleFontSize,
      ytitleFontSize = ytitleFontSize,
      titleFontSize = titleFontSize,
      bgcol = bgcol,
      opacity = opacity,
      axisCol = axisCol
    )
  )
}
